db/db_impl/db_impl_open.cc

```diff
@@ -1580,7 +1579,9 @@ IOStatus DBImpl::CreateWAL(uint64_t log_file_num, uint64_t recycle_log_number,
         tmp_set.Contains(FileType::kWalFile)));
     *new_log = new log::Writer(std::move(file_writer), log_file_num,
                                immutable_db_options_.recycle_log_file_num > 0,
-                               immutable_db_options_.manual_wal_flush);
+                               immutable_db_options_.manual_wal_flush,
+                               immutable_db_options_.wal_compression);
+    io_s = (*new_log)->AddCompressionTypeRecord();
```

Comment:
Could this be done by the log::Writer constructor, hiding the details from the caller? The failure handling would be a bit tricky. Maybe it can store the status and return it on the next record addition.

Reply:
Done.